lib/socket.dart

```diff
@@ -120,7 +128,7 @@ class SocketIO {
   }
 
   /// checks whether connection is alive
-  Future<bool> isConnected() => _channel.invokeMethod(
+  Future<bool?> isConnected() => _channel.invokeMethod(
```

Comment:
If we can implement a method that ensures returning of null-safe values like the one [i created here](https://github.com/ably/ably-flutter/blob/3fabe8bc0e921984b932319ccd340694c8931447/lib/src/platform/src/platform.dart#L50) would avoid usage of unwanted `?` and `!`